core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/StreamBatchSource.java

```diff
@@ -94,7 +96,9 @@
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     super.configurePipeline(pipelineConfigurer);
     streamBatchConfig.validate();
-    pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
+    if (!streamBatchConfig.containsMacro(Properties.Stream.NAME)) {
+      pipelineConfigurer.addStream(new Stream(streamBatchConfig.name));
```

Comment:
you mean because cdap doesn't allow stream creation at runtime?  I think its still a valid use case to have a macro as the stream name, it just means that the stream will have to be created by somebody else. We should mention that in the plugin docs though.